grpc/rust/BUILD

```diff
@@ -16,13 +16,15 @@
 #
```

Comment:
The previous build architecture, with a `cargo_build_script` at its core, would not have led to a good UX. The idea was (following Tonic's own docs) for `typedb-protocol` to come bundled with the `.proto` sources and a build script that would allow the end-user to compile the protobufs on-the-fly. This would require them to have `protoc` installed in their environment, and would dramatically increase their compilation time and overhead.

Instead, we needed a way to bundle the generated Rust sources. First, we created a `rust_binary` that ran the build script. But this was a `bazel run` target; we needed a `build` target. So now, we've created a rule, `rust_tonic_compile`, that internally runs this `rust_binary` (similarly to how many of our rules run Kotlin binaries), and exposes the outputs. Any `rust_library` can then depend on these generated sources. This is similar to the behaviour of the `antlr` rule, and the API is similar to Java and Python (`java_grpc_compile` and `python_grpc_compile` respectively)